python/graphstorm/tracker/sagemaker_tracker.py

```diff
@@ -52,7 +42,7 @@ def log_metric(self, metric_name, metric_value, step, force_report=False):
         force_report: bool
             If true, report the metric
         """
-        if force_report or self._do_report(step):
```

Comment:
do_report is useful when we are sending log to a system like mlflow or tensorboard.

Reply:
SageMaker Tracker also use this to decide whether to print the train step.

Reply:
I see. We have to keep the argument for the `do_report` argument from SageMaker. For GSF, we can just make it an internal argument by disabling `group.add_argument("--log-report-frequency",...)`

Reply:
What is our final decision for the "--log-report-frequency"?

Reply:
I would say, let's keep it for now.

Reply:
OK. Closing this PR.